### PR TITLE
Fixed bundling

### LINKED ISSUE #60: @emotion/react when it is already loaded.
Hi,

How are you?
We are using this great package, but we notice that when we are working locally we receive this yellow warning message in the console.
We tried to update @emotion libraries but it's not solved the problem.

/node_modules/react-timezone-select/dist/esm/index.js
**You are loading @emotion/react when it is already loaded. Running multiple instances may cause problems. This can happen if multiple versions are used, or if multiple builds of the same version are used.**

I hope you can assist us with this manner.

### PR DESCRIPTION
**Pull request by DreierF**
### Description

The library is currently published as a full bundle including react, react-dom, emotion, ... (see [source at unpkg](https://unpkg.com/browse/react-timezone-select@1.3.2/dist/esm/index.js)). Libraries are however not supposed to package their third-party dependencies. That job should be left to the bundler of the final application. The rollup setup even attempts to do that by applying the `rollup-plugin-peer-deps-external` at least for the `peerDependencies`, which seems to have no effect (might be the order in which the plugins are applied, but not sure).

This PR switches to [tsup](https://github.com/egoist/tsup), which I would consider the de-facto standard tool to publish libraries written in Typescript to npm. It also automatically provides the correct behavior of not bundling the third-party dependencies.

Let me know what you think 🙂 

### Linked Issues

Should fix #60

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,29 +55,25 @@
     "timezone-soft": "^1.3.1"
   },
   "devDependencies": {
-    "@rollup/plugin-commonjs": "^21.0.1",
-    "@rollup/plugin-node-resolve": "^13.1.3",
-    "@rollup/plugin-typescript": "^8.3.0",
+    "@babel/core": "^7.20.7",
```
Comment (DreierF) [inline package.json:58]: Fixes the pnpm error about missing peer dependencies

File: src/tests/TimezoneSelect.spec.tsx [hunk 2]
```diff
@@ -28,6 +28,7 @@ test('loads and displays default timezone - passing full object', async () => {
       value={{
         value: 'Europe/Amsterdam',
         label: '(GMT+1:00) Amsterdam, Berlin, Bern, Rome, Stockholm, Vienna',
+        offset: 60,
```
Comment (DreierF) [inline src/tests/TimezoneSelect.spec.tsx:31]: Fixes a Typescript error as offset is required here

**Comment by ndom91** [comment]
Hey sorry its taken forever to get around to this. Looks great though, thanks!

**Comment by ndom91** [comment]
Published in `1.5.1` :pray:

**Pull request merged**